src/main/java/uk/gov/hmcts/darts/transcriptions/mapper/TranscriptionResponseMapper.java

```diff
@@ -404,23 +405,41 @@ public TranscriptionDocumentHideResponse mapHideOrShowResponse(TranscriptionDocu
         response.setIsHidden(entity.isHidden());
 
         if (objectAdminActionEntity != null) {
-            AdminActionResponse adminActionResponse = new AdminActionResponse();
-            adminActionResponse.setId(objectAdminActionEntity.getId());
-            adminActionResponse.setReasonId(objectAdminActionEntity.getObjectHiddenReason().getId());
-            adminActionResponse.setHiddenById(objectAdminActionEntity.getHiddenBy().getId());
-            adminActionResponse.setHiddenAt(objectAdminActionEntity.getHiddenDateTime());
-            adminActionResponse.setIsMarkedForManualDeletion(objectAdminActionEntity.isMarkedForManualDeletion());
-            adminActionResponse.setMarkedForManualDeletionById(objectAdminActionEntity.getMarkedForManualDelBy().getId());
-            adminActionResponse.setMarkedForManualDeletionAt(objectAdminActionEntity.getMarkedForManualDelDateTime());
-            adminActionResponse.setTicketReference(objectAdminActionEntity.getTicketReference());
-            adminActionResponse.setComments(objectAdminActionEntity.getComments());
+            AdminActionResponse adminActionResponse = mapAdminActionResponse(objectAdminActionEntity);
+            response.setAdminAction(adminActionResponse);
+        }
+
+        return response;
+    }
 
+    public AdminApproveDeletionResponse mapAdminApproveDeletionResponse(TranscriptionDocumentEntity entity, ObjectAdminActionEntity objectAdminActionEntity) {
+        AdminApproveDeletionResponse response = new AdminApproveDeletionResponse();
+        response.setId(entity.getId());
+        response.setIsHidden(entity.isHidden());
+
+        if (objectAdminActionEntity != null) {
+            AdminActionResponse adminActionResponse = mapAdminActionResponse(objectAdminActionEntity);
             response.setAdminAction(adminActionResponse);
         }
 
         return response;
     }
 
+    private AdminActionResponse mapAdminActionResponse(ObjectAdminActionEntity objectAdminActionEntity) {
```

Comment:
Could use MapStrut

Reply:
Following the same pattern used in this class